src/psyclone/dynamo0p3.py

```diff
@@ -6549,12 +6548,17 @@ def __init__(self, parent=None, loop_type=""):
             elif self.loop_type == "colour":
                 tag = "cell_loop_idx"
                 suggested_name = "cell"
-            elif self.loop_type == "dofs":
+            elif self.loop_type == "dof":
                 tag = "dof_loop_idx"
                 suggested_name = "df"
-            else:
+            elif self.loop_type == "":
                 tag = "cell_loop_idx"
                 suggested_name = "cell"
+            else:
+                raise InternalError(
+                    "Unsupported loop type '{0}' supplied. Supported values "
+                    "are 'colours', 'colour', 'dof' or '' (for cell-columns).".
```

Comment:
What about "null" in the `VALID_LOOP_TYPES` above? If this is supposed to be in the list as well then I would suggest using the list for the error message here.

Reply:
No, this is just for the loops that have a variable. I've updated the error message to make this clearer.

Reply:
Thanks for the explanation!